src/prefect/results.py

```diff
@@ -472,25 +474,23 @@ class PersistedResult(BaseResult):
 
     type: str = "reference"
 
-    serializer_type: str
     storage_key: str
+    serializer: Serializer
```

Comment:
Moved `serializer` to a public attr to ensure it is included when marshaling this model to JSON during transaction record persistence.